build/patches/Enable-crash-reporting-ui.patch

```diff
@@ -43,30 +162,131 @@ diff --git a/chrome/browser/ui/webui/crashes_ui.cc b/chrome/browser/ui/webui/cra
  
    bool system_crash_reporter = false;
  #if BUILDFLAG(IS_CHROMEOS_ASH) || BUILDFLAG(IS_CHROMEOS_LACROS)
-@@ -244,11 +243,6 @@ void CrashesDOMHandler::HandleRequestSingleCrashUpload(
+@@ -244,12 +261,78 @@ void CrashesDOMHandler::HandleRequestSingleCrashUpload(
    bool success = args->GetString(0, &local_id);
    DCHECK(success);
  
 -  // Only allow manual uploads if crash uploads aren’t disabled by policy.
 -  if (!ChromeMetricsServiceAccessor::IsMetricsAndCrashReportingEnabled() &&
 -      IsMetricsReportingPolicyManaged()) {
 -    return;
--  }
-   upload_list_->RequestSingleUploadAsync(local_id);
++  base::ThreadPool::PostTaskAndReplyWithResult(
++      FROM_HERE, kLoadingTaskTraits,
++      base::BindOnce(&CrashesDOMHandler::RequestSingleUpload, base::Unretained(this), local_id),
++      base::BindOnce(&CrashesDOMHandler::RequestSingleUploadCallback, base::Unretained(this), local_id));
++}
++
++std::string CrashesDOMHandler::RequestSingleUpload(const std::string& local_id) const {
++  // get crash file path
++  std::string info_file_path = upload_list_->GetFilePathByLocalId(local_id);
++  if (info_file_path.empty()) {
++    LOG(ERROR) << "Crash report: file path is not set for " << local_id;
++    return std::string();
++  }
++  base::FilePath crash_file_path(info_file_path);
++
++  // get android crash report dir
++  base::FilePath cache_dir;
++  base::android::GetCacheDirectory(&cache_dir);
++  base::FilePath upload_log_path = cache_dir.Append("Crash Reports");
++
++  // crash reports can have multiple extensions (e.g. foo.dmp, foo.dmp.try1,
++  // foo.skipped.try0), remove it
++  base::FilePath zip_file = crash_file_path;
++  while (zip_file != zip_file.RemoveExtension())
++    zip_file = zip_file.RemoveExtension();
++
++  // make zip file name, like "ec708a7b-cb17-44e7-8dae-e32f6c45cb8c.zip"
++  zip_file = upload_log_path.Append(zip_file.BaseName())
++                            .AddExtensionASCII(".zip");
++  // if exists return it
++  if (base::PathExists(zip_file))
```

Comment:
Why would it already exist?

Reply:
as the download is done from the ui download on which I have no feedback, it may happen that the user cancels the download.
in this case, I still allow to retry the download and in that case the zip is already present.

Reply:
Can you update the comment to explain this?